### PR TITLE
feat(queue): Spotify support

### PR DESCRIPTION
**Pull request by PikaProgram**
Title, using [spotify-url-info](https://www.npmjs.com/package/spotify-url-info) package to fetch data and [ytsr](https://www.npmjs.com/package/ytsr) + [ytdl](https://www.npmjs.com/package/ytdl) to search and download the track from youtube.

- [x] Resolve Dependencies
- [x] Figure How Pipe Data And Play The Track
- [x] Implementation
- [x] Bugs Mashing & Code Refactoring

ps: would prob start next week cz of exams

### CODE REVIEW

**Comment by mzrtamp** [comment]
*Don't review yet* ? Why don't you convert to draft within' this PR

File: src/commands/PlayCommand.ts [hunk 1]
```diff
@@ -76,6 +78,21 @@ export class PlayCommand extends BaseCommand {
...
             }
         }
+
+        if (/^https?:\/\/(open\.spotify\.com)\/playlist(.*)$/g.exec(url)) {
+            try {
+                const tracks = await getTracks(url);
+                const playlistTitles: string[] = tracks.map((track: { artists: { name: string }[]; name: string }): string => `${track.artists[0].name} - ${track.name}`);
+                playlistTitles.map(async title => {
+                    const videoResults = await this.client.youtube.searchVideos(title, this.client.config.searchMaxResults);
+                    const queuedVideo = await this.client.youtube.getVideo(videoResults[0].id);
```
Comment (mzrtamp) [inline src/commands/PlayCommand.ts:88]: 'queuedVideo' is assigned a value but never used

Comment (PikaProgram) [reply src/commands/PlayCommand.ts:88]: "dont review yet" :v

Comment (mzrtamp) [reply src/commands/PlayCommand.ts:88]: :apasih:

**Comment by mzrtamp** [comment]
Hello @PikaProgram, it looks like someone is doing the same thing with you, and it looks like he's already done it and is no longer in the process, do we have to just approve that one over there? I suggest it like that, and you can try to take care of SoundCloud instead of Spotify

**Comment by PikaProgram** [comment]
Y :apasih:

**Pull request closed without merging**